native-sql-engine/cpp/src/precompile/gandiva.h

```diff
@@ -40,13 +40,7 @@ T round2(T val, int precision = 2) {
 }
```

Comment:
looks like this round2 func can also utilize below new cast method?

Reply:
Is arrow::Decimal128 stored internally using some char-like stuffs? I didn't see a "Reprecision" method or something is provided so I think it might be OK for just now to use string formatters. By the way I see there is another function `std::setprecision` might be easier to use than snprintf.